dali/tensorflow/daliop.cc

```diff
@@ -49,11 +49,18 @@ namespace tf = tensorflow;
       }                                                                            \
     } while (0)
 
-tf::TensorShape DaliToShape(int64_t* ns) {
+template <typename T>
+using AutoCPtr = std::unique_ptr<T, decltype(std::free)*>;
```

Comment:
Nitpick 1: you're allocating with `malloc`, but freeing with `std::free`
Nitpick 2: I think that under some circumstances `decltype(std::free)` might be something other than a function (e.g. a function pointer). In this case, the code breaks.
Nitpick 3: Since you're always using `std::free`, it'd be more efficient to have an empty deleter class that always calls free:
```
struct CDeleter {
  void operator()(void *p) { free(p); }
};
template <typename T>
using AutoCPtr = std::unique_ptr<T, CDeleter>();
```